File: src/longHourAndAHalf/ConsoleUI.kt
package longHourAndAHalf

/**
 * Command line game interface.
 * @author JavaBird
 *
 * @property gameplay Gameplay object to get the game data from.
 */
class ConsoleUI(override val gameplay: Gameplay) : UI {
    override fun doTurn(slideText: SlideText, actions: List<Action>): Action {
        printTurnData(slideText, actions)
        return askAction(actions)
    }

    private fun printTurnData(slideText: SlideText, actions: List<Action>) {
        with(gameplay.character) {
            println("Character: $name")
            println("Gender: $gender")
            println()
            println("Urine in bladder: ${bladder.urine} ml")
            println("Pee holding power: ${bladder.sphincterPowerPercent}")
            println("Incontinence: ${bladder.incontinence}x")
            println("Underwear: ${underwear.color} $underwear")
            println("Outerwear: ${outerwear.color} $outerwear")
            println("Underwear capacity: ${underwear.capacity}")
            println("Outerwear capacity: ${outerwear.capacity}")
            printSeparator()
            println(slideText)
            printSeparator()
            printActions(actions)
        }
    }

    private fun printActions(actions: List<Action>) {
        for (i in actions.indices)
            println("${i + 1}: ${actions[i]}")
    }

    private fun printSeparator() {
        repeat(50, { print('_') })
        println()
    }

    private fun askAction(actions: List<Action>): Action {
        print("> ")
        var actionNumber: Int?
        do
            actionNumber = readLine()?.toInt()
        while (actionNumber == null || actionNumber !in 1..actions.size)
        printSeparator()
        return actions[actionNumber - 1]
    }
}

File: src/ui/console_ui.py
from action import Action
from ui.game_ui import UI


class ConsoleUI(UI):
    def __print_characteristics(self):
        with self.__gameplay as g:
            print(g.character.name)
            print('Bladder fullness:', g.character.bladder.urine, 'ml')
            print('Water in tummy:', g.character.tummy_water, 'ml')
            print('Sphincter strength: ', g.character.bladder.sphincter.strength, '%', sep='')
            print('Clothes dryness:', g.character.dryness)
            print('Embarrassment:', g.character.embarrassment)
            print('Incontinence: ', g.character.bladder.sphincter.incontinence, 'x', sep='')
            print('Time:', g.world.time)
            print('Underwear:', g.character.underwear)
            print('Outerwear:', g.character.outerwear)

    def do_turn(self, text, actions) -> Action:
        print(text)
        print()
        self.__print_characteristics()
